templates/cluster/main.go

```diff
@@ -2,6 +2,7 @@
 package main
```

Comment:
I think the file needs to be:

`// package main holds the implementation of the cluster template.
package main

import (
	"context"
	"log"
	"time"

	"github.com/nextmv-io/sdk/cluster/kmeans"
	"github.com/nextmv-io/sdk/measure"
	"github.com/nextmv-io/sdk/run"
)

type input struct {
	Points        []measure.Point `json:"points"`
	Clusters      int             `json:"clusters"`
	Weights       []int           `json:"weights"`
	MaximumWeight int             `json:"maximum_weight"`
	MaximumPoints int             `json:"maximum_points"`
}

type cluster struct {
	Index    int             `json:"index"`
	Centroid measure.Point   `json:"centroid"`
	Points   []measure.Point `json:"points"`
	Indices  []int           `json:"indices"`
}

// Output is the output of the solver.
type Output struct {
	Clusters          []cluster       `json:"clusters"`
	Feasible          bool            `json:"feasible"`
	Unassigned        []measure.Point `json:"unassigned"`
	UnassignedIndices []int           `json:"unassigned_indices"`
}

func main() {
	err := run.CLI(solver).Run(context.Background())
	if err != nil {
		log.Fatal(err)
	}
}

type ClusterOptions struct {
	Duration time.Duration `json:"duration" default:"10s"`
	Clusters int           `json:"clusters" default:"2"`
}

func solver(input input, opts ClusterOptions) ([]Output, error) {
	// We start by creating a kmeans model.
	model, err := kmeans.NewModel(input.Points, input.Clusters)
	if err != nil {
		return nil, err
	}

	// We set the maximum weight and maximum points for each cluster.
	for _, c := range model.ClusterModels() {
		c.SetMaximumPoints(input.MaximumPoints)

		_, err = c.SetMaximumSumValue(input.MaximumWeight, input.Weights)
		if err != nil {
			return nil, err
		}
	}

	// We create a solver with the model.
	solver, err := kmeans.NewSolver(model)
	if err != nil {
		return nil, err
	}

	// We create the solve options we will use and set the time limit
	// and the measure to use.
	solveOptions := kmeans.NewSolveOptions().
		SetMeasure(measure.EuclideanByPoint()).
		SetMaximumDuration(opts.Duration)

	solution, err := solver.Solve(solveOptions)
	if err != nil {
		panic(err)
	}

	return []Output{format(solution)}, nil
}

// format returns a function to format the solution output.
func format(solution kmeans.Solution) Output {
	o := Output{
		Clusters:          make([]cluster, len(solution.Clusters())),
		Feasible:          solution.Feasible(),
		Unassigned:        solution.Unassigned(),
		UnassignedIndices: solution.UnassignedIndices(),
	}

	for idx, c := range solution.Clusters() {
		o.Clusters[idx] = cluster{
			Index:    idx,
			Centroid: c.Centroid(),
			Points:   c.Points(),
			Indices:  c.Indices(),
		}
	}

	return o
}
`